### PR TITLE
pidstore: proper HTTP error if PID is already exists

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: backend/inspirehep/pidstore/errors.py [hunk 1]
```diff
@@ -20,3 +24,18 @@ class MissingControlNumber(BaseInspirePidStoreError):
...
+
+
+class PIDAlreadyExistsError(PIDAlreadyExists, PIDError):
+    code = 400
+    description = "PIDAlreadyExists: pid_type:'{pid_type}', pid_value:'{pid_value}'."
+
+    def __init__(self, pid_type, pid_value, **kwargs):
+        super().__init__(pid_type=pid_type, pid_value=pid_value, **kwargs)
+        self.description = self.description.format(
+            pid_type=pid_type, pid_value=pid_value
```
Comment (pazembrz) [inline backend/inspirehep/pidstore/errors.py]: Didn't we decided to use fstrings?
`description = lambda: f"PIDAlreadyExists: pid_type:'{pid_type}', pid_value:'{pid_value}'."

and self.description=self.description()` or smthng like that?

Comment (drjova) [reply backend/inspirehep/pidstore/errors.py]: yes indeed, I just put the template as a static. Anyway changed it :)

File: backend/inspirehep/pidstore/errors.py [hunk 2]
```diff
@@ -20,3 +24,17 @@ class MissingControlNumber(BaseInspirePidStoreError):
...
+    pass
+
+
+class PIDAlreadyExistsError(PIDAlreadyExists, PIDError):
+    code = 400
+
+    def __init__(self, pid_type, pid_value, **kwargs):
+        super().__init__(pid_type=pid_type, pid_value=pid_value, **kwargs)
+        self.description = (
+            "PIDAlreadyExists: pid_type:'{pid_type}', pid_value:'{pid_value}'."
```
Comment (pazembrz) [inline backend/inspirehep/pidstore/errors.py]: **F**-string? ;)

Comment (drjova) [reply backend/inspirehep/pidstore/errors.py]: :tired_face:

**Pull request merged**